biorxiv/word_vector_experiment/biorxiv_generate_embeddings.py

```diff
@@ -0,0 +1,279 @@
+
+    for idx, article in tqdm_notebook(document_df.iterrows()):
+        tree = (
+                ET.parse(
+                    open(f"../biorxiv_articles/{article['document']}", "rb"),
+                    parser = parser
+                )
+            )
+        root = tree.getroot()
+            
+        word_vectors = []
+        abstract_text = root.xpath("//abstract/*//text()")
+            
+        word_vectors += [
+            list(
+                map(
+                    lambda x: model.wv[x], 
+                    filter(
+                        lambda x: x in model.wv, 
+                        text.split(" ")
+                    )
+                )
+            )
+            for text in abstract_text
+        ]
+            
+        abstract_vectors = (
+            list(
+                itertools.chain.from_iterable(
+                    filter(lambda x: len(x) > 0, word_vectors)
+                )
+            )
+        )
+            
+        article_text = root.xpath("//body/sec/*//text()")
+            
+        word_vectors += [
+            list(
+                map(
+                    lambda x: model.wv[x], 
+                    filter(
+                        lambda x: x in model.wv, 
+                        text.split(" ")
+                    )
+                )
+            )
+            for text in article_text
+        ]
+            
+        article_vectors = (
+            list(
+                itertools.chain.from_iterable(
+                    filter(lambda x: len(x) > 0, word_vectors)
+                )
+            )
+        )
+            
+        total_vectors = abstract_vectors + article_vectors 
+
+        # skips weird documents that don't contain text
+        if len(total_vectors) > 0:
+            document_vec_map[article['document']] = pd.np.stack(total_vectors).mean(axis=0)
+
+    return document_vec_map
+
+
+# In[ ]:
+
+
+for word_model_path in Path().rglob("output/word2vec_models/*/*.model"):
+    model_dim = word_model_path.parents[0].stem
+    
+        
+    word_model = Word2Vec.load(str(word_model_path.resolve()))
+    biorxiv_vec_map = generate_doc_vectors(word_model, journal_map_df, skip_methods=False)
+
+    biorxiv_vec_df = pd.DataFrame([
+        [document] + biorxiv_vec_map[document].tolist()
+        for document in biorxiv_vec_map
+        ], 
+        columns=["document"] + list(map(lambda x: f"feat_{x}", range(int(model_dim))))
+    )
+    
+    biorxiv_vec_df.to_csv(
+        f"output/word2vec_output/biorxiv_all_articles_{model_dim}.tsv.xz", 
+        sep="\t", index=False,
+        compression="xz"
+    )
+
+
+# # UMAP the Documents
+
+# After generating document embeddings, the next step is to visualize all the documents. In order to visualize the embeddings a low dimensional representation is needed. UMAP is an algorithm that can generate this representation, while grouping similar embeddings together. 
```

Comment:
Is this set of cells redundant to the other notebook?

Reply:
>Is this set of cells redundant to the other notebook?

No. Actually, this set of cells saves me time by generating all necessary embeddings for the visualize biorxiv experiment. I grouped all embedding generating code into one notebook and the other notebooks are designed to visualize the embeddings. This designed made the most sense in my head.